### DISCUSSION TITLE
External packages missing nova 4 support?

### DISCUSSION THREAD

**Discussion by LTKort** [laravel/nova-issues#4217] [General]
We've been busy updating a lot of our packages to support Nova v4. 

Is there a package that you use that is missing support for Nova 4? Maybe an abandoned package?

We like to help upgrading the package to support Nova v4!

**Comment by RhysLees** [comment]
Make sure to add Nova ^4.0 in required so it is tagged on novapsckages.com

**Comment by mrleblanc101** [comment]
Nova sortable doesn't support Nova 4 and it's pretty important for our workflow

**Reply by LTKort** [reply]
Yeah, we have the same. 
I have been busy fixing it, but it won’t work since issue #4143 is still open.

**Reply by LTKort** [reply]
Have a PR open right now: https://github.com/outl1ne/nova-sortable/pull/121#issue-1234017605

**Comment by philipbaginski** [comment]
Can you take a look at this package: https://github.com/zareismail/voice-recorder

**Reply by LTKort** [reply]
What does it do? What’s the goal and what does not work right now? There is no  documentation and there are no issues

**Comment by dmason30** [comment]
I have unmerged Nova 4.x compatibility PRs on the following packages if you are open to maintaining any of them it would be greatly appreciated! We are using all of these as forks in real projects and they work great!

https://github.com/coroo/nova-chartjs/pull/129 - merged
https://github.com/superlatif-io/nova-tag-input/pull/18
https://github.com/sadekd/nova-opening-hours-field/pull/23 - merged

**Reply by LTKort** [reply]
Hi @dmason30 

- I will check the ChartJs package, we use our own fork for a client as well. 
- Instead of the tag package I would use https://github.com/outl1ne/nova-multiselect-field package which supports a taggable() function that achieves the same goal and is well maintained.
- I will check out the Opening hours package when I have some extra time, I think I can combine it with a private package I’ve built recently

**Reply by dmason30** [reply]
We already use the multi-select field unfortunately we need a tag field where you let the user add any value (not restricted by options). Which is supported by the above tag field but not the multi-select field.

**Reply by dmason30** [reply]
Update the opening hours one has been merged and tagged 🥳

**Reply by dmason30** [reply]
I guess I timed this wrong ( i did wait 3 weeks 🤣 ) but the chartjs one just got merged and tagged too!

**Comment by ianrobertsFF** [comment]
[nova-flexible-content](https://github.com/whitecube/nova-flexible-content) had an initial PR for Nova4 functionality, but there's still stuff broken, some with PR's that fix the issues. Would probably be handy if it had a new maintainer, but its not a small package to maintain.

**Reply by LTKort** [reply]
What is still not working? We have our own fork of it that works well

**Reply by ianrobertsFF** [reply]
DependsOn within flexible fields doesnt work by default. And similarly neither do Slug fields.

I have a PR in (https://github.com/whitecube/nova-flexible-content/pull/349) but the required trait from my PR is frankly hacky, due to the way Nova gets the fields for the PATCH requests. I personally can't see a better way to achieve it but I dont know the codebases for Nova or Flexible Content well enough to state that there is no better way. 

The field itself doesn't support dependsOn

Also the way it implements some translations break nova translations: https://github.com/whitecube/nova-flexible-content/issues/350

**Reply by ianrobertsFF** [reply]
Just FYI was testing my dependsOn PR against the latest nova and it needed some fixes, trait is a bit more robust now too.

**Comment by ibrunotome** [comment]
That one would be great! https://github.com/tighten/nova-google-analytics

**Comment by kylescousin** [comment]
Would be great if this Nova Mega Filter could be made compatible with v4: https://github.com/dcasia/nova-mega-filter

It worked up until v3.

**Comment by philipbaginski** [comment]
Very usefull package: https://github.com/outl1ne/nova-detached-filters

Please take a look at it :-)

**Comment by sutcliffe121** [comment]
I am completing a Nova 3 to 4 upgrade and I've had to remove the packages below as they aren't Nova 4 compatible and I haven't been able to find an alternative yet:

- benjaminhirsch/nova-slug-field
- davidpiesse/nova-map
- benjacho/belongs-to-many-field
- dillingham/nova-attach-many

Also: 

- kraftbit/nova-tinymce5-editor -> I could use: https://novapackages.com/packages/mostafaznv/nova-ckeditor
- classic-o/nova-media-library -> I could use: https://novapackages.com/packages/ebess/advanced-nova-media-library

I used the following packages in Nova 3 but have been able to switch to Nova 4 native features (which is great news!):

- epartment/nova-dependency-container -> the native functionality is not as good as this package, but I have it mostly working apart from on actions
- laraning/nova-time-field
- maatwebsite/laravel-nova-excel -> CSV is fine for my use cases
- pdewit/nova-external-url
- jackabox/nova-duplicate-field

**Reply by LTKort** [reply]
We do have a package for:
Tinymce: https://github.com/marshmallow-packages/nova-tinymce
Media library: https://github.com/marshmallow-packages/advanced-nova-media-library

these are forks and fixed for nova 4

**Comment by vesper8** [comment]
Would love to see this package receive some Nova 4 love https://github.com/iMuMz/NovaMapCard

I tried to implement it myself using https://github.com/vue-leaflet/vue-leaflet which supports Vue 3, but I'm having all sorts of troubles and getting errors such as `runtime-core.esm-bundler.js:226 ChunkLoadError: Loading chunk node_modules_leaflet_dist_leaflet-src_esm_js failed.` which I can't seem to be able to solve. Can't even get a basic demo working.

**Reply by carestad** [reply]
DId you get leaflet working in Nova 4?

**Reply by vesper8** [reply]
Nope, after trying and running into issues I pivoted and used these instead:

```

    "vue3-gmap-custom-marker": "^0.0.2",
    "vue3-google-map": "^0.11.0",
```

**Reply by carestad** [reply]
Ah, too bad.

I kind of gave up and went along with

```php
Nova::remoteStyle('https://unpkg.com/leaflet@1.9.2/dist/leaflet.css');
Nova::remoteScript('https://unpkg.com/leaflet@1.9.2/dist/leaflet.js');
```

so I could at least re-use most of my logic and already existing leaflet based code. Not 100% ideal, but it works I guess.

**Reply by scramatte** [reply]
Hi,

I get it working as necessary. The problem is with dynamic chunks path loading. You need to tweak webpack config and make accessible these chunks with some routes.  I've currently made an api route (/nova-vendor/nova-leaflet/chunks/{filename}) that returns file contents of js chunk. It's still not very clean, I need to understand the way that should expose these assets Nova.  Documentation is not very detailed about this. I hope to be able to publish  it soon.

Regards

**Reply by iMuMz** [reply]
A Nova 4 compatible package has been released 

https://github.com/iMuMz/Nova4CardMap

**Comment by kylescousin** [comment]


**Reply by mrleblanc101** [reply]
These are third party package, it's not up to Nova team to maintain them... This thread has been created so volunteer could find the packages that are not compatible and open PR. If you are not happy, volunteer your own time.

**Reply by ianrobertsFF** [reply]
Sure, probably benefit everyone if the nova devs built all of their projects for them too.

Third party packages are just that. Third Party. No support from the Nova team. If there is a Nova 3 package you absolutely need but its no longer maintained you could always update it and submit a PR or Fork it.

**Reply by mrleblanc101** [reply]
Nova is not subscription based, I don't know why you would make such things up. 

Also your analogy make no sense. This is like saying Google should fix and develop Microsoft Android app because it makes them money. No... They make a public API that 3rd party dev can use.

**Reply by kylescousin** [reply]
When you replied, I had already edited that comment. You do get Nova updates only up to a certain version. If you have v3 and you want v4 => Pay again. 

Then you're stuck with third-party packages that don't work anymore. Your analogy comparing a niche product like Nova with Android, which has millions of apps with completely different codebases, makes no sense. The Nova developers know the Nova code inside out and when they upgrade to a new version, it would take them a small effort to help upgrade the most used packages to remain compatible.

This would benefit everyone (Nova team, package creators, and customers).

Anyway... *unsubscribe*

**Reply by mrleblanc101** [reply]
Stop wasting everyone's time if you don't understand how open source work @kylescousin 

It's exactly the same thing, Nova provide the APIs, if you don't want compatibility problem, don't use 3rd party plugin.

Otherwise, it's up to the package maintainer to update them, if they can't, please contribute YOUR time with a PR or find an alternative package.

**Comment by vesper8** [comment]
Would love to see https://github.com/saintsystems/nova-linkable-metrics gain Nova 4 support

**Comment by chrillep** [comment]
- [x] @brandonferens - https://github.com/kirschbaum-development/nova-comments/issues/47
  - [x] PR - https://github.com/kirschbaum-development/nova-comments/pull/50
- [ ] @manogi - https://github.com/kraenkvisuell/nova-astrotranslatable/issues/5
- [ ] @the-alichemist - https://github.com/gobrightspot/nova-detached-actions/issues/69
- [ ] @yassilah - https://github.com/yassilah/laravel-nova-nested-form/issues/162
- [ ] @NikolaySav - https://github.com/NikolaySav/nova-percent-field/issues/5
- [x] @milewski -https://github.com/dcasia/custom-relationship-field/pull/11

**Reply by brandonferens** [reply]
Thanks for the reminder @chrillep!

**Reply by chrillep** [reply]
nova-comments merged https://github.com/kirschbaum-development/nova-comments/pull/50  . thank you @adammparker for the feedback! 🙏🏼

**Reply by chrillep** [reply]
[custom-relationship-field](https://github.com/dcasia/custom-relationship-field) merged,  thanks @milewski for the help! ❤️

**Comment by mrleblanc101** [comment]
Either https://github.com/nikaia/nova-rating-field or https://github.com/KosmosKosmos/nova-rating would be nice in Nova 4

**Comment by antoniosiles** [comment]
Would love to see https://novapackages.com/packages/sloveniangooner/searchable-select gain Nova 4 support

**Comment by andypooletrioteca** [comment]
This package is full of bugs and seems to have been abandoned:

https://github.com/eminiarts/nova-tabs

It's quite important for complex nova sites.

**Comment by oluomotoso** [comment]
Would be great if you could help update this, I have a fork but haven't been able to get it to work. https://github.com/fourstacks/nova-repeatable-fields

**Comment by clementmas** [comment]
I was using https://github.com/markrassamni/nova-resource-landing-page on Nova 3.

Now I'm not sure how to accomplish this in Nova 4.
